src/types.ts

```diff
@@ -0,0 +1,108 @@
+import type { JITI } from "jiti";
+import type { JITIOptions } from "jiti/dist/types";
+import type { DotenvOptions } from "./dotenv";
+
+export interface ConfigLayerMeta {
+  name?: string;
+  [key: string]: any;
+}
+
+export type UserInputConfig = Record<string, any>;
+
+export interface C12InputConfig<
+  T extends UserInputConfig = UserInputConfig,
+  MT extends ConfigLayerMeta = ConfigLayerMeta
+> {
+  $test?: T;
+  $development?: T;
+  $production?: T;
```

Comment:
here's how it could be done:

```suggestion
export type C12InputConfig<
  T extends UserInputConfig = UserInputConfig,
  MT extends ConfigLayerMeta = ConfigLayerMeta,
  Env extends string = 'test' | 'development' | 'production',
> = Record<`$${Env}`, T> & {
  $env?: Record<string, T>;
  $meta?: MT;
}
```

Reply:
Thanks for the example. But i like to know more about why we need such customization in first place. Because if it is common enough, i don't see any issue that we directly support (or unsupport) them from built-in aliases. But if we do support this, it adds more complexity and less usage consistency across ecosystem and possible breaking changes once introducing a runtime check to only support a sub-set of aliases.

Reply:
For sure, if you plan to support the three listed and don't want to allow users to configure, then we should leave it as is. I was proposing adding at this stage only to avoid a type-breaking change later down the line if we later propose supporting this kind of feature.

Reply:
(resolved in discord discussion) We might rethink to do this later if needed.